src/BizHawk.Client.EmuHawk/tools/Watch/WatchValueBox.cs

```diff
@@ -31,18 +32,16 @@ public WatchSize ByteSize
 				if (changed)
 				{
 					SetMaxLength();
-
-					var isTypeCompatible = value switch
+					var validTypes = value switch
 					{
-						WatchSize.Byte => ByteWatch.ValidTypes.Any(t => t == _type),
-						WatchSize.Word => WordWatch.ValidTypes.Any(t => t == _type),
-						WatchSize.DWord => DWordWatch.ValidTypes.Any(t => t == _type),
-						_ => false,
+						WatchSize.Byte => ByteWatch.ValidTypes,
+						WatchSize.Word => WordWatch.ValidTypes,
+						WatchSize.DWord => DWordWatch.ValidTypes,
+						_ => [ ],
 					};
-
-					if (!isTypeCompatible)
+					if (!validTypes.Contains(_type))
 					{
-						_type = WatchDisplayType.Unsigned;
+						_type = validTypes.Count is 0 ? WatchDisplayType.Unsigned : validTypes[1];

```

Comment:
Typo? Should be `validTypes[0]` instead? Also the above switch should probably include `WatchSize.QWord` now.